src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,191 @@
+    }),
+
+    source: positional({
+      type: string,
+      displayName: 'path',
+      description: 'path to source data including collection.json file',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    if (args.targetBucketName.startsWith('s3://')) {
+      throw new Error(
+        `target-bucket-name entered as path, did you mean ${args.targetBucketName
+          .replace('s3://', '')
+          .replace('/', '')}?`,
+      );
+    }
+
+    logger.info({ source: args.source }, 'GeneratePath:Start');
+
+    const collection = await fsa.readJson<StacCollection>(fsa.join(args.source, 'collection.json'));
+    if (collection == null) throw new Error(`Failed to get collection.json from ${args.source}.`);
+
+    const tiff = await loadFirstTiff(args.source, collection);
+
+    const category = getCategory(collection);
+    const geographicDescription = getGeographicDescription(collection);
+    const region = getRegion(collection);
+    const event = getEvent(collection);
+    const date = getDate(collection);
+    const gsd = extractGsd(tiff);
+    const epsg = extractEpsg(tiff);
+
+    const target = generatePath(args.targetBucketName, category, geographicDescription, region, event, date, gsd, epsg);
+    logger.info({ duration: performance.now() - startTime, target: target }, 'GeneratePath:Done');
+    return target;
+  },
+});
+
+export function generatePath(
+  targetBucketName: string,
+  category: string,
+  geospaital_description: string,
+  region: string,
+  event: string,
+  date: string,
+  gsd: string,
+  epsg: string,
+): string {
+  const name = generateName(region, geospaital_description, event);
+
+  if (category === dataCategories.SCANNED_AERIAL_PHOTOS) {
+    // nb: Historic Imagery is out of scope as survey number is not yet recorded in collection metadata (15/02/24)
+    throw new Error(`Automated target generation not implemented for historic imagery`);
+  } else if ([dataCategories.URBAN_AERIAL_PHOTOS, dataCategories.RURAL_AERIAL_PHOTOS].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if (category === dataCategories.SATELLITE_IMAGERY) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if ([dataCategories.DEM, dataCategories.DSM].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}/${category}_${gsd}/${epsg}/`;
+  } else {
+    throw new Error(`Path Can't be generated from collection as no matching category: ${category}.`);
+  }
+}
+
+export function generateName(region: string, geospaital_description: string, event: string): string {
+  /* Function to:
+    - Determine if region or geospatial description should be the dataset name
+    - Add event to name
+  */
+  let name = region;
+  if (geospaital_description) {
+    name = geospaital_description.toLowerCase().replace(/\s+/g, '-');
```

Comment:
Can we use the [`slugify` function](https://github.com/linz/argo-tasks/pull/827) here?

Reply:
Yep, this would be a good spot for that, I just needed this workaround to get the tests to work and be real examples

Reply:
I have now added the slugify function